listener.go

```diff
@@ -61,12 +64,10 @@ func (l *listener) handleIncoming() {
 
 	var catcher tec.TempErrCatcher
 	for {
-
-		select {
-		case <-l.ticket:
-		case <-l.ctx.Done():
-			return
-		}
+		// no need to wait on the context.
+		// Close will drain the incoming channel which will cause the
```

Comment:
Nit: The result is correct, but the reason is not. When the context is canceled, the select in line 110 selects the `ctx.Done()` case, and then `Release` decrements the counter.

Reply:
Ah. Correct.